core/src/main/java/io/codiga/utils/TreeSitterUtils.java

```diff
@@ -53,65 +57,62 @@ public static Optional<AstElement> getAstElement(
 
   public static Optional<TreeSitterAstElement> getFullAstTree(String code, Language language) {
     Optional<Long> treeSitterLanguage = languageToTreeSitterLanguage(language);
-    TreeCursor treeCursor;
     if (treeSitterLanguage.isEmpty()) {
       return Optional.empty();
     }
 
-    try(Parser parser = new Parser()) {
+    try (Parser parser = new Parser()) {
       parser.setLanguage(treeSitterLanguage.get());
-      Tree tree = parser.parseString(code);
-      return getTreeFromNode(tree.getRootNode());
+      try (Tree tree = parser.parseString(code)) {
+        return getTreeFromNode(tree.getRootNode());
+      }
     } catch (UnsupportedEncodingException e) {
       logger.info("error when decoding the code");
       return Optional.empty();
     }
   }
 
-    public static Optional<TreeSitterAstElement> getTreeFromNode(Node node) {
-        TreeCursor treeCursor = node.walk();
-        var visitedChildren = false;
-        var isFinished = false;
-        TreeSitterAstElement current = null;
-        TreeSitterAstElement parent = null;
+  public static Optional<TreeSitterAstElement> getTreeFromNode(Node node) {
+    try (TreeCursor treeCursor = node.walk()) {
```

Comment:
Is the `try` necessary here, since we call `treeCursor.close()` on line 114?

Reply:
Yes because it automatically close if not already done for whatever reason